src/crypto/sha256.cpp

```diff
@@ -720,7 +720,7 @@ std::vector<unsigned char> CSHA256::Save() const {
 
     WriteLE64(&result[32], bytes << 3);
 
-    memcpy(&result[40], buf, buf_size);
```

Comment:
Based on my reading here: https://stackoverflow.com/questions/29844298/is-it-legal-to-call-memcpy-with-zero-length-on-a-pointer-just-past-the-end-of-an 
I modified the implementation for a check for `memcpy` 0 bytes. 
Apparently `memcpy(src, object_end_ptr, 0)` is illegal and undefined behaviour. 

Added a separate commit for highlighting this part. Will squash later if needed.

Reply:
You can just say `if (buf_size)`, but this is also okay.